plugins/dfu/dfu-target.c

```diff
@@ -198,6 +198,12 @@ dfu_target_parse_sector (DfuTarget *target,
 		return FALSE;
 	}
 
+  // MATEKF722SE board quirk, sector type is the first char, the second char is empty
+  if (tmp[1] == '\0') {
+      tmp[1] = tmp[0];
+      tmp[0] = 'B';
+  }
+
```

Comment:
Can you explain a bit on why this happens? Is it the kind of thing we should add a fwupd quirk for? It seems slightly dangerous to assume 'B' in all cases. Also, tabs please.

Reply:
I agree; this should be done as a quirk on this particular board.

Reply:
I am not sure why this happens, but this certainly isn't documented by dfu protocol. 
I copied this assumption from dfu-util https://github.com/Stefan-Schmidt/dfu-util/blob/fdc72a8a738994dc72cfd63ff08f690e5145277e/src/dfuse_mem.c#L170.

i agree this, should be a quirk, i can implement new quirk, but can you give me general guides or point me to example of similar quirk in the code?

Reply:
At a high level add a new quirk type here: https://github.com/fwupd/fwupd/blob/02930f1f23e67c410c808d7ead6dd7da6982c461/plugins/dfu/dfu-device.c#L45

Add an entry into `dfu.quirk` for your device and the quirks it needs (via Flags key).

And then in the code look for that quirk and activate the changed behavior.

Just force push this branch with the fixes once you have them done.

Reply:
ok, i added absent-sector-size